packages/transactions/src/AddGatewayV1.ts

```diff
@@ -17,23 +19,40 @@ interface SignOptions {
 
 export default class AddGatewayV1 extends Transaction {
   public owner?: Addressable
+
   public gateway?: Addressable
+
   public payer?: Addressable
+
   public ownerSignature?: Uint8Array
+
   public gatewaySignature?: Uint8Array
+
   public payerSignature?: Uint8Array
+
   public stakingFee?: number
+
   public fee?: number
 
   constructor(opts: AddGatewayOptions) {
     super()
+
     this.owner = opts.owner
     this.gateway = opts.gateway
     this.payer = opts.payer
     this.stakingFee = 0
     this.fee = 0
-    this.fee = this.calculateFee()
-    this.stakingFee = Transaction.stakingFeeTxnAddGatewayV1
+
```

Comment:
I think we'll also want to deserialize the signatures and add those to the constructor if they're provided. that way, when we deserialize the partial txn that the hotspot creates, we keep the signature it added to it

Reply:
one note is to assign them in the constructor after the `fee` calc because that calculation wipes out the signatures in order to perform its calculation